packages/react/src/template.tsx

```diff
@@ -259,7 +263,13 @@ export function BaseTemplate({
       )
     }
     if (isStateDataState(propValue)) {
-      return render(getLocalStateValue(propValue), path)
```

Comment:
bug: this would previously throw an error if state was an object. we should call "renderProp" not "render" so that if it's an object or an array (e.g. nested with other state) this will work.